src/components/dashboard/FaceRecognition/FaceRecognition.js

```diff
@@ -0,0 +1,185 @@
+// @flow
+import FRapi from './FaceRecognitionAPI'
+import ZoomCapture from './ZoomCapture'
+import zoomSdkLoader from './ZoomSdkLoader'
+import React, { createRef } from 'react'
+import { type ZoomCaptureResult } from './Zoom'
+import { StyleSheet, View } from 'react-native'
+import GDStore from '../../../lib/undux/GDStore'
+import type { DashboardProps } from '../Dashboard'
+import logger from '../../../lib/logger/pino-logger'
+import { Wrapper, CustomButton, Section } from '../../common'
+import normalize from 'react-native-elements/src/helpers/normalizeText'
+
+const log = logger.child({ from: 'FaceRecognition' })
+
+type FaceRecognitionProps = DashboardProps & {
+  screenProps: any,
+  store: Store
+}
+
+type State = {
+  showPreText: boolean,
+  showZoomCapture: boolean,
+  loadingFaceRecognition: boolean,
+  loadingText: string,
+  facemap: Blob,
+  zoomReady: boolean,
+  captureResult: ZoomCaptureResult
+}
+
+/**
+ * Responsible to orchestrate FaceReco process, using the following modules: ZoomCapture & FRapi.
+ * 1. Loads ZoomCapture and recieve ZoomCaptureResult - the user video capture result after processed locally by ZoomSDK (Handled by ZoomCapture)
+ * 2. Uses FRapi which is responsible to communicate with GoodServer and UserStorage on FaceRecognition related actions, and handle sucess / failure
+ * 3. Display relevant error messages
+ * 4. Enables/Disables UI components as dependancy in the state of the process
+ **/
+class FaceRecognition extends React.Component<FaceRecognitionProps, State> {
+  state = {
+    showPreText: true,
+    showZoomCapture: false,
+    loadingFaceRecognition: false,
+    loadingText: '',
+    facemap: null,
+    zoomReady: false,
+    captureResult: null
+  }
+
+  containerRef = createRef()
+  width = 720
+  height = 0
+
+  componentWillUnmount = () => {
+    zoomSdkLoader.unload()
+  }
+
+  componentWillMount = async () => {
+    this.loadedZoom = await zoomSdkLoader.load()
+    if (this.loadedZoom) {
+      this.timeout = setTimeout(() => {
+        this.zoomReady = true // TODO: handle zoom init issues.
+      }, 500)
+    }
+  }
+
+  componentDidMount = async () => {
+    this.setWidth()
+    this.timeout && clearTimeout(this.timeout)
```

Comment:
clearing timeout here will prevent the timeout to complete.
You want to clear the timeout at the component will unmount for the case where the user immediatly goes back so you want to cleanup

Reply:
you are right, that's was supposed to be on componentUnmount

Reply:
part of master